src/toast/ops/mapmaker_utils/kernels.py

```diff
@@ -92,7 +92,7 @@ def build_noise_weighted(
         weight_index,
         weights,
         data_index,
-        det_data,
+        det_data.astype(np.float64),
```

Comment:
obviously no harm here, though we could also enforce the up-cast during HDF5 loading.  We just have to remember to find all these places again once we have the compiled functions templated over the types.